### PR TITLE
Base layer

### PR DESCRIPTION
**Pull request by badnames**
The old branch/pull request had far too many changes and grew to an unmanageable size.

## Basics

- [x] Details of what you changed are in commit messages.
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildserver is happy.
- [x] The PR is rebased with current master.

## Checklist

- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I fixed all affected decisions
- [ ] I added unit tests for my code
- [ ] I added code comments, logging, and assertions as appropriate
- [ ] I translated all strings visible to the user
- [ ] I mentioned [every code or binary](/.reuse/dep5) not directly written or done by me in [reuse syntax](https://reuse.software/)
- [ ] I created left-over issues for things that are still to be done
- [ ] Code is conforming to [our Architecture](/doc/architecture)
- [ ] Code is conforming to [our Guidelines](/doc/guidelines)
      (exceptions are documented)
- [ ] Code is consistent to [our Design Decisions](/doc/decisions)

## Review

- [ ] I've tested the code
- [ ] I've read through the whole code
- [ ] Examples are well chosen and understandable

### CODE REVIEW

**Comment by markus2330** [comment]
Unfortunately, I see "Die Baselayer-Datei konnte nicht geladen werden!" when being logged in on https://pr.permaplant.net and using a file that should exist. Am I doing something wrong?

**Comment by markus2330** [comment]
As said, more specific error messages would be very helpful. @absurd-turtle how is the progress for a file selector? Then some type of errors would be more obvious.

**Comment by badnames** [comment]
> Unfortunately, I see "Die Baselayer-Datei konnte nicht geladen werden!" when being logged in on https://pr.permaplant.net and using a file that should exist. Am I doing something wrong?

Might be a CORS issue, but I have to look into it. How can I access the relevant branch on pr.permaplant.net?

**Comment by markus2330** [comment]
jenkins build please

**Comment by markus2330** [comment]
If nobody pushed it is still there, otherwise it will be again there after the build finishes. See also doc/ci.md

**Comment by badnames** [comment]
I added some code that produces more detailed error messages.

After trying to load the image, the following error was thrown:
"Error loading base layer: TypeError: Url must be a string. Received undefined"

This error also occurs in https://pr.permaplant.net/webdav while trying to upload an image.
Ergo: this does not seem to be an issue with the base layer.

@markus2330, @lukashartl @absurd-turtle Is the Nextcloud base URL set correctly on pr.permaplant.net?

File: frontend/src/components/Form/SimpleFormInput.tsx [hunk 1]
```diff
@@ -1,82 +1,37 @@
-import { HTMLInputTypeAttribute } from 'react';
 import { FieldValues, Path, UseFormRegister } from 'react-hook-form';
 
-interface SimpleFormInputProps<T extends FieldValues> {
+interface SimpleFormInputProps<T extends FieldValues>
+  extends React.InputHTMLAttributes<HTMLInputElement> {
```
Comment (markus2330) [inline frontend/src/components/Form/SimpleFormInput.tsx:4]: Please add documentation at least for new elements.

File: frontend/src/config/i18n/de/baseLayer.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+  "loadingFailed": "Fehler beim laden des Baselayers: ",
```
Comment (markus2330) [inline frontend/src/config/i18n/de/baseLayer.json:2]:
```suggestion
  "loadingFailed": "Fehler beim Laden des Baselayers: ",
```

File: frontend/src/config/i18n/de/baseLayer.json [hunk 3]
```diff
@@ -0,0 +1,4 @@
...
+  "notAnImage": "Der gewählte Baselayer Hintergrund ist keine Bilddatei!"
```
Comment (markus2330) [inline frontend/src/config/i18n/de/baseLayer.json:3]:
```suggestion
  "notAnImage": "Der gewählte Baselayer Hintergrund ist leider keine Bilddatei."
```

**Review by markus2330 — CHANGES_REQUESTED** [review]
> "Error loading base layer: TypeError: Url must be a string. Received undefined"

I can only confirm, I get the same error. And no frog on the map. :cry:

**Comment by markus2330** [comment]
Finally it is done, good work! :1st_place_medal: :heart: :partying_face: 

Let us see if it works on dev.permaplant.net (see also  #486)

@badnames Please create follow-up issues as necessary so that we clearly see what is missing in the base layer UC.

**Pull request merged**